src/librustc_mir/borrow_check/move_errors.rs

```diff
@@ -371,18 +371,7 @@ impl<'a, 'gcx, 'tcx> MirBorrowckCtxt<'a, 'gcx, 'tcx> {
                     }) => true,
                     _ => false,
                 };
-                if try_remove_deref && snippet.starts_with('*') {
-                    // The snippet doesn't start with `*` in (e.g.) index
-                    // expressions `a[b]`, which roughly desugar to
-                    // `*Index::index(&a, b)` or
-                    // `*IndexMut::index_mut(&mut a, b)`.
-                    err.span_suggestion(
-                        span,
-                        "consider removing the `*`",
-                        snippet[1..].to_owned(),
-                        Applicability::Unspecified,
-                    );
-                } else {
+                if !try_remove_deref || !snippet.starts_with('*') {
```

Comment:
I think this condition should be removed and we should accept that sometimes we will suggest `&*`

Reply:
I've made that change. You said we should accept that sometimes we will suggestion strange things, but it's worth highlighting some of the weirder:

https://github.com/rust-lang/rust/commit/132aa6d861ba29cdd37d85d9d885505711d60496#diff-33c6ee1b44f5207df102942b0c659b94R26

https://github.com/rust-lang/rust/commit/132aa6d861ba29cdd37d85d9d885505711d60496#diff-a3121673048182c8d1a0ee3ccbfb9d5fR8

Are those suggestions acceptable? They might end up causing more confusion than the initial suggestion this PR addresses.

Reply:
The first is fine. The `&*&` suggestions are weird, but they need someone to write `*&`. I think that we are generally OK with suggestions that are suboptimal in edge cases. cc @rust-lang/wg-diagnostics

Reply:
It'd be nice to add the comment `// run-rustfix` at the beginning of the changed tests (and rerun `x.py`) to make sure the suggestions are always valid in these cases and do not break going forwards.

> I think that we are generally OK with suggestions that are suboptimal in edge cases.

We have a harder stance on _incorrect_ and _misleading_ suggestions over _weird_ suggestions, but we do try to minimize the later :)

Reply:
Suggestions from borrow checking are always going to be best guesses. They definitely shouldn't be rustfixable, unless rustfix has a flag to apply possibly incorrect suggestions.

Reply:
Hi, just wanted to confirm -- I should not do Esteban's suggestion?

Reply:
@kennethbgoodin it's fine, don't bother for now. I'll try to clean it up sometime down the line :)